ops/main.py

```diff
@@ -369,97 +365,191 @@ def _should_use_controller_storage(db_path: Path, meta: CharmMeta) -> bool:
         return False
 
 
-def main(charm_class: Type[ops.charm.CharmBase],
-         use_juju_for_storage: Optional[bool] = None):
-    """Setup the charm and dispatch the observed event.
+class _Abort(Exception):  # noqa: N818
+    """Raised when something happens that should interrupt ops execution."""
 
-    The event name is based on the way this executable was called (argv[0]).
+    def __init__(self, exit_code: int, **kwargs: Any):
+        super().__init__(*kwargs)
```

Comment:
Any reason the kwargs forwarding is needed here?
```suggestion
    def __init__(self, exit_code: int):
        super().__init__()
```